File: ListaPOO_01/ex03.py
# Entidade
class ContaBancaria:
    def __init__(self):
        self.titular = "sem nome"
        self.numero = "sem número"
        self.__saldo = 0              # Encapsulamento
    def depositar(self, valor):
        if valor >= 0: 
            self.__saldo += valor
        else:
            #print("Valor deve ser positivo")   
            raise ValueError("Valor deve ser positivo")  
    def sacar(self, valor):
        if valor >= 0 and self.__saldo >= valor:
            self.__saldo -= valor
        else:
            #print("Saldo Insuficiente")     
            raise ValueError("Valor deve ser positivo ou saldo insuficiente")  
    def consultar_saldo(self):
        return self.__saldo

# Interface com o usuário
x = ContaBancaria()
x.titular = "Eduardo"
x.numero = "123-0"

#print(x.saldo)     saldo da conta está encapsulado
#print(x.__saldo)

print(x.titular, x.numero, x.consultar_saldo())
x.depositar(1000)
print(x.titular, x.numero, x.consultar_saldo())
x.sacar(1200)
print(x.titular, x.numero, x.consultar_saldo())

File: ListaPOO_01/ex03.java
import java.util.Scanner;

public class ex03 {
    public static void main(String[] args) {
        Scanner Scanner = new Scanner(System.in);
        int opt = 0;
        Conta conta = new Conta(Scanner.next(), Scanner.nextInt(), Scanner.nextDouble());
        while (opt != 4) {
            System.out.println("1 - Depositar");
            System.out.println("2 - Sacar");
            System.out.println("3 - Consultar Saldo");
            System.out.println("4 - Sair");
            opt = Scanner.nextInt();
            switch (opt) {
                case 1:
                    System.out.println("Digite o valor a ser depositado: ");
                    conta.depositar(Scanner.nextDouble());
                    break;
                case 2:
                    System.out.println("Digite o valor a ser sacado: ");
                    conta.sacar(Scanner.nextDouble());
                    break;  
                case 3:
                    System.out.println("Saldo: " + conta.consultarSaldo());
                    break;  
                case 4: 
                    System.out.println("Saindo...");
                    break;
        }
    }
}
}